logger.go

```diff
@@ -79,3 +78,45 @@ func Sublogger(inp ZapCompatibleLogger, subname string) (loggerRet ZapCompatible
 
 	return loggerRet
 }
+
+// LogWith adds fields for logging to a given ZapCompatibleLogger instance.
+// This function uses reflection to dynamically add fields to the provided logger by
+// calling its `WithFields` method if it is an RDK logger, or its `With` method if it is a Zap logger.
+// If neither method is available, it logs a debug message and returns the original logger.
+func LogWith(inp ZapCompatibleLogger, args ...interface{}) (loggerRet ZapCompatibleLogger) {
```

Comment:
[nit] I find the name of this function _slightly_ confusing. It's more adding the fields to `inp`, while `LogWith` for some reason, to me, implies there's some "global" logger we're now "logging with". Could we call this `AddFieldsToLogger`? Open to counterarguments/other naming suggestions, too.

Reply:
I like `AddFieldsToLogger` enough, I kept the word With to be consistent with the functions it calls, but definitely not necessary